AK/DefaultDelete.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+namespace AK {
+
+template<class T>
+struct DefaultDelete {
+    constexpr DefaultDelete() noexcept = default;
+
+    template<class U>
+    constexpr DefaultDelete(DefaultDelete<U> const& d) noexcept { }
```

Comment:
So are these actually needed? I don't see where the copy constructor would be instantiated, and I'd be very curious why not having these would cause a compile error now, but not a linker error before

Reply:
Removing it would make the `DefaulDelete` object default copy/move constructible. The defaults are not `constexpr`. That is ok with me since they are not used at the moment, but they might get used in the future and I wouldn't want to prevent `constexpr` capabilities. If you think it is better to leave them out until they are needed I can remove them.

Reply:
I think a copy constructor from a different type than `T` doesn't really make sense, unless we also slap a requirement on there for `IsConvertible<U*, T*>`.

But adding unused code is a bit unneeded IMO, if anything I would `= delete` the copy/move constructor. I can't think of a scenario where these would even need to be copied/moved, but if someone needs it in the future, they'll have a better understanding of the required semantics.

We also don't need any of the `noexcept` qualifiers, Serenity does not use exceptions.